### PR TITLE
Refactor Custom Serving Runtime admin

### LINKED ISSUE #1225: [Feature Request]: Custom Serving Runtimes -- Admins Should Have Access
### Feature description

Custom Serving Runtimes uses the k8s pass through API -- this is a problem for basic "Admins".

We need close this gap about the settings page and the admins. We either need to grant them access to the dashboard namespace fully or stick the effort behind the service account (this is not ideal).

Today a basic user that is an Admin of our dashboard cannot view/modify the Custom Serving Runtimes.

### Describe alternatives you've considered

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by lucferbux**
## Description

Closes #1225 

## How Has This Been Tested?

For all testing, the pre-requisite is:
1. Add a regular user to the odh admins group (you should have some groups added in Settings -> User Management)
2. Log in as the regular user
3. You should see now the admin panel

*Note*: In dev mode it takes way more time to do any action compared to pass through

### Fetch
1. Go to **Settings -> Serving Runtime**
2. You should see all the Custom Serving Runtimes

Screenshot 2023-05-29 at 14 09 51

### Arrange
1. Drag and drop the Custom Serving Runtimes
2. Navigate to other screen
4. The order should persist

### Create
1. Click on "Add serving runtime"
2. Add a new serving runtime
3. Check that you can create it

Screenshot 2023-05-29 at 14 19 28

### Delete
1. Go back to the Serving Runtime view
2. Click on the kebab menu of one of the Serving Runtimes
3. Click delete
5. Check that you can delete it

Screenshot 2023-05-29 at 14 17 00

## Test Impact

All these tests should be covered in the serving runtime testing efforts

## Request review criteria:

- [x] The commits have meaningful messages (squashes happen on merge by the bot).
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

### CODE REVIEW

**Review by Gkrumbach07 — APPROVED** [review]
everything works except i get this error when changing the order

```
Error updating the serving runtimes order
Template order is not configured
```

but i get that error for both user types so i think it is just not configured right

**Comment by lucferbux** [comment]
> everything works except i get this error when changing the order
> 
> 
> 
> ```
> 
> Error updating the serving runtimes order
> 
> Template order is not configured
> 
> ```
> 
> 
> 
> but i get that error for both user types so i think it is just not configured right

Yeah, that's intended, this happens when you don't have configured the templaOrder field in the dashboard config CRD

**Comment by lucferbux** [comment]
@DaoDaoNoCode @alexcreasy Can someone review this too please?

**Review by DaoDaoNoCode — APPROVED** [review]
tested, works good, code lgtm too!

File: frontend/src/pages/modelServing/customServingRuntimes/DeleteCustomServingRuntimeModal.tsx [hunk 1]
```diff
@@ -35,7 +35,8 @@ const DeleteCustomServingRuntimeModal: React.FC<DeleteCustomServingRuntimeModalP
       onDelete={() => {
         if (template) {
           setIsDeleting(true);
-          deleteTemplate(template.metadata.name, template.metadata.namespace)
+          // TODO: Revert back to pass through api once we migrate admin panel
+          deleteTemplateBackend(template.metadata.name, template.metadata.namespace)
```
Comment (shalberd) [inline frontend/src/pages/modelServing/customServingRuntimes/DeleteCustomServingRuntimeModal.tsx:39]: so by passing it though the api, you use the serviceaccount privileges?  Are you planning to extend the Openshift default ClusterRoles for project rolebindings, e.g Admin, View, Edit with custom Clusterroles, e.g. an Admin being able to delete Custom Serving Runtimes or to list storageclasses? Or do you go with the standard logged-in user roles  Admin, View, Edit ?

**Pull request merged**